src/simpledb/plan/BasicQueryPlanner.java

```diff
@@ -2,49 +2,54 @@
 
 import java.util.*;
 import simpledb.tx.Transaction;
+import simpledb.materialize.*;
 import simpledb.metadata.*;
 import simpledb.parse.*;
 
 /**
  * The simplest, most naive query planner possible.
+ * 
  * @author Edward Sciore
  */
 public class BasicQueryPlanner implements QueryPlanner {
    private MetadataMgr mdm;
-   
+
    public BasicQueryPlanner(MetadataMgr mdm) {
       this.mdm = mdm;
    }
-   
+
    /**
-    * Creates a query plan as follows.  It first takes
-    * the product of all tables and views; it then selects on the predicate;
-    * and finally it projects on the field list. 
+    * Creates a query plan as follows. It first takes the product of all tables and
+    * views; it then selects on the predicate; and finally it projects on the field
+    * list.
     */
    public Plan createPlan(QueryData data, Transaction tx) {
-      //Step 1: Create a plan for each mentioned table or view.
-     List<Plan> plans = new ArrayList<>();
+      // Step 1: Create a plan for each mentioned table or view.
+      List<Plan> plans = new ArrayList<>();
       for (String tblname : data.tables()) {
          String viewdef = mdm.getViewDef(tblname, tx);
          if (viewdef != null) { // Recursively plan the view.
             Parser parser = new Parser(viewdef);
             QueryData viewdata = parser.query();
             plans.add(createPlan(viewdata, tx));
-         }
-         else
+         } else
             plans.add(new TablePlan(tx, tblname, mdm));
       }
-      
-      //Step 2: Create the product of all table plans
+
+      // Step 2: Create the product of all table plans
       Plan p = plans.remove(0);
       for (Plan nextplan : plans)
          p = new ProductPlan(p, nextplan);
-      
-      //Step 3: Add a selection plan for the predicate
+
+      // Step 3: Add a selection plan for the predicate
       p = new SelectPlan(p, data.pred());
-      
-      //Step 4: Project on the field names
+
+      // Step 4: Project on the field names
       p = new ProjectPlan(p, data.fields());
+
+      // Step 5: Sort on the records
+      p = new SortPlan(tx, p, data.sortfields());

```

Comment:
think this has to be optional
<img width="1392" alt="Screenshot 2022-02-13 at 2 30 51 AM" src="https://user-images.githubusercontent.com/41518656/153723712-d3d8afa6-075f-471e-802e-d504cb4b158c.png">
l